src/main/java/bgu/spl/net/impl/stomp/ConnectionsImpl.java

```diff
@@ -0,0 +1,66 @@
+package bgu.spl.net.impl.stomp;
+
+import bgu.spl.net.srv.ConnectionHandler;
+import bgu.spl.net.srv.Connections;
+
+import java.util.AbstractMap;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.ConcurrentHashMap;
+
+public class ConnectionsImpl implements Connections<Frame> {
+
+    private ConcurrentHashMap<Integer, ConnectionHandler<Frame>> _connections;
+    // SimpleEntry used as a Pair object
+    private ConcurrentHashMap<String, List<AbstractMap.SimpleEntry<ConnectionHandler<Frame>, Integer>>> _channels;
+
+    public ConnectionsImpl() {
+        _connections = new ConcurrentHashMap<>();
+        _channels = new ConcurrentHashMap<>();
+    }
+
+    @Override
+    public boolean send(int connectionId, Frame msg) {
+        try {
+            _connections.get(connectionId).send(msg);
+        }
+        catch (Exception e) {
+            return false;
+        }
+        return true;
+    }
+
+    private Frame addSubscriptionId(Frame msg, Integer id) {
+        Frame f = msg.clone();
+        f.addHeader("subscription", id.toString());
+        return f;
+    }
+
+    @Override
+    public void send(String channel, Frame msg) {
+        for(AbstractMap.SimpleEntry<ConnectionHandler<Frame>, Integer> sub : _channels.get(channel)) {
+            sub.getKey().send(addSubscriptionId(msg, sub.getValue())); // Send the message with the subscription header
```

Comment:
is Connetction handler had a .send function? why not use this.send(int, Frame)?

Reply:
a. ConnectionHandler has a send function, thats how this.send(int, Frame) works.
b. you are right I should use it. I will change it